src/side-nav/side-nav-item.tsx

```diff
@@ -0,0 +1,82 @@
+import { useContext, useEffect, useState } from "react";
+import {
+    Container,
+    DefaultButton,
+    TitleContainer,
+} from "./side-nav-item.styles";
+import { SideNavItemProps } from "./types";
+import { SideNavContext } from "./side-nav.context";
+
+export const SideNavItem = ({
+    icon,
+    title,
+    onClick,
+    ...otherProps
+}: SideNavItemProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const id = otherProps.id || title.toLowerCase().replaceAll(" ", "-");
+    const { selectedItem, setSelectedItem } = useContext(SideNavContext);
+    const [highlight, setHighlight] = useState<boolean>();
+    const [selected, setSelected] = useState<boolean>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        if (otherProps.selected) {
+            setSelectedItem(id);
+        }
+    }, []);
+
+    useEffect(() => {
+        if (selectedItem === id && !selected) {
+            setSelected(true);
+            setHighlight(true);
+        } else if (selected) {
+            setSelected(false);
+            setHighlight(false);
+        }
+    }, [selectedItem]);
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleMouseEnter = () => {
+        if (!highlight && !selected) {
+            setHighlight(true);
+        }
+    };
+
+    const handleMouseLeave = () => {
+        if (highlight && !selected) {
+            setHighlight(false);
+        }
+    };
+
+    const handleOnClick = () => {
+        if (selected) return;
+        setSelectedItem(id);
+        if (onClick) {
+            onClick();
+        }
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    return (
+        <Container
```

Comment:
This Container itself could be a plain button containing a styled icon and styled text. This would improve accessibility as the icon is associated with a text. You can also handle the hover state via css with `:hover`.